gnocchi/carbonara.py

```diff
@@ -85,6 +84,46 @@ def round_timestamp(ts, freq):
         (ts - UNIX_UNIVERSAL_START64) / freq) * freq
 
 
+def make_timeseries_array(array):
+    """Create a Numpy array with the right format for a timeseries."""
+    return numpy.array(array,
+                       dtype=[('timestamps', 'datetime64[ns]'),
+                              ('values', 'float64')])
+
+
+def make_timeseries(timestamps, values):
+    """Return a Numpy array representing a timeseries.
+
+    This array specifies correctly the data types, which is important for
+    Numpy to operate fastly.
+    """
+    l = len(timestamps)
+    if l != len(values):
+        raise ValueError("Timestamps and values must have the same length")
+    arr = make_timeseries_array(numpy.zeros(l))
```

Comment:
Yes, this is to make it have the right shape. You can't assign otherwise as the shape would be  (0,) and the values/timestamps would be (number of points,).

I did not find a way to create directly a `numpy.array` based on 2 array. I might be able to use records instead, I'll look into that.